### PR TITLE
Hotfix/dppb 663 rules search

### PR DESCRIPTION
**Pull request by Gabriel-Popescu**
## As a contributor
By submitting this PR I confirm I have read and understood the Github Standards document and understand the role I play in ensuring standards, security, and assurance at DfT.

## As a PR reviewer
By accepting this PR I understand my responsibilities as described in the the Github standards document.

# Proposed changes

Describe the big picture of your changes here to make it clear what your pull request will change. Make sure to explain the **what** e.g. the functionality it changes not the **how** e.g. what code you have written.  If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Documentation Update (change to naming or other documentation)

## Points I have checked in my code

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR._

- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have checked that my changes have not broken any other functionality
- [ ] I have linked to any issues this PR fixes

## Points for review

Add checkboxes here for any specific aspects of your PR you would like to be checked in peer review. If you don't specify anything here, the reviewer will check:

- [ ] All of your unit tests pass
- [ ] Your code is clear and easy to understand
- [ ] You have documented any new features
- [ ] They can run the code you have written
- [ ] Your changes do not break any other functionality

## Security considerations

_Put an `x` in the boxes that apply. You should have considered all of these points before submitting a PR._

- [ ] My code does not contain any secrets such as passwords, API keys, etc
- [ ] I have not accidentally uploaded any data to Github
- [ ] My code does not include any personal information such as names or email addresses

## Who is reviewing this PR?

Tag your reviewer here to ensure they get a notification!

### CODE REVIEW

File: Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs [hunk 1]
```diff
@@ -66,16 +66,86 @@ public DtroResponse MapToDtroResponse(Models.DataBase.DTRO dtro)
...
         foreach (Models.DataBase.DTRO dtro in dtros)
         {
-            List<ExpandoObject> regulations = dtro.Data.GetValueOrDefault<IList<object>>("Source.provision")
-                .OfType<ExpandoObject>()
-                .SelectMany(it => it.GetValue<IList<object>>("regulation").OfType<ExpandoObject>())
-                .ToList();
+            Console.WriteLine("MapToSearchResult-DTRO:");
+            PrindDtroData(dtro.Data);
+            Console.WriteLine("Before List<ExpandoObject> regulations");
+//            List<ExpandoObject> regulations = dtro.Data.GetValueOrDefault<IList<object>>("Source.provision")
```
Comment (VinP-PA) [inline Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs:109]: Please get rid of this comment section

File: Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs [hunk 2]
```diff
@@ -66,16 +66,86 @@ public DtroResponse MapToDtroResponse(Models.DataBase.DTRO dtro)
...
     {
         List<DtroSearchResult> results = new List<DtroSearchResult>();
-
         foreach (Models.DataBase.DTRO dtro in dtros)
         {
-            List<ExpandoObject> regulations = dtro.Data.GetValueOrDefault<IList<object>>("Source.provision")
-                .OfType<ExpandoObject>()
-                .SelectMany(it => it.GetValue<IList<object>>("regulation").OfType<ExpandoObject>())
-                .ToList();
+            Console.WriteLine("MapToSearchResult-DTRO:");
```
Comment (VinP-PA) [inline Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs:106]: Can you please remove these Console.WriteLines in this method?

File: Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs [hunk 3]
```diff
@@ -66,16 +66,86 @@ public DtroResponse MapToDtroResponse(Models.DataBase.DTRO dtro)
...
+            PrindDtroData(dtro.Data);
```
Comment (VinP-PA) [inline Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs:107]: Please remove this line

File: Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs [hunk 4]
```diff
@@ -66,16 +66,86 @@ public DtroResponse MapToDtroResponse(Models.DataBase.DTRO dtro)
...
+                        continue;
+                    }
+
+                    var expandoRegulations = regulationList.OfType<ExpandoObject>();
+                    regulations.AddRange(expandoRegulations);
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"An error occurred: {ex.Message}");
```
Comment (VinP-PA) [inline Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs:140]: Keep this Console.WriteLine

File: Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs [hunk 5]
```diff
@@ -275,20 +345,31 @@ public void InferIndexFields(ref Models.DataBase.DTRO dtro)
...
-            OrderReportingPoint = dtro.OrderReportingPoints,
-            RegulationStart = regulationStartDates,
-            RegulationEnd = regulationEndDates,
-            Id = dtro.Id
-        };
+    private DtroSearchResult CopyDtroToSearchResult(Models.DataBase.DTRO dtro, List<DateTime> regulationStartDates, List<DateTime> regulationEndDates)
+    {
+        DtroSearchResult result = new DtroSearchResult();
+        result.TroName = dtro.Data.GetValueOrDefault<string>("Source.troName");
+        Console.WriteLine($"Tro name:\t {result.TroName}");
```
Comment (VinP-PA) [inline Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs:352]: Can you please remove these Console.WriteLines in this method?

File: Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs (hunk 1 above)
Comment (Gabriel-Popescu) [reply Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs:109]: done

File: Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs (hunk 2 above)
Comment (Gabriel-Popescu) [reply Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs:106]: done

File: Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs (hunk 3 above)
Comment (Gabriel-Popescu) [reply Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs:107]: done

File: Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs (hunk 5 above)
Comment (Gabriel-Popescu) [reply Src/DfT.DTRO/Services/Mapping/DtroMappingService.cs:352]: done

**Review by VinP-PA — APPROVED** [review]

**Pull request merged**